pkg/syncer/spec/mutators/secrets.go

```diff
@@ -48,18 +49,23 @@ func (sm *SecretMutator) Mutate(downstreamObj *unstructured.Unstructured) error
 		return err
 	}
 
-	// We need to transform the default kcp token into an Opaque secret, in order to avoid the pcluster to rewrite it.
-	if secret.Name == "kcp-default-token" {
+	// We need transform the serviceaccount tokens into an Opaque secret, in order to avoid the pcluster to rewrite them,
+	// and we remove the annotations that point to the kcp serviceaccount name/uid.
+	if _, ok := secret.Annotations[core.ServiceAccountNameKey]; ok {
```

Comment:
why decode the secret? Nothing in here needs that.

Reply:
done